### DISCUSSION TITLE
ASP.NET CORE

### DISCUSSION THREAD

**Discussion by depaula36** [balta-io/forum#598] [Tira Dúvidas] [answered]
Professor,
Para aplicações ASP.NET CORE (minha ideia é fornecer html através do padrão MVC no futuro), como levantar a própria máquina sem servidor externo (IIS) para prover o serviço?
Pelo que entendi, há opções através do Kestrel ou Weblistener que já viriam com ASP.NET CORE. _Como faria a configuração para um weblistener para o .Net8_?
Dúvida adicional, conseguiria testar em casa a 'conexão com esse host'? Ou dependeria de ter um IP Fixo para testar? (não sei nada de rede..rsrsrs).
Desde já, agradeço

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @depaula36, como vai?

Para aplicações ASP.NET CORE (minha ideia é fornecer html através do padrão MVC no futuro), como levantar a própria máquina sem servidor externo (IIS) para prover o serviço?
- Não recomendo usar sua máquina como hospedagem (Nem o IIS 🤣), mas o próprio comando `dotnet run` já executa o projeto como um servidor stand alone (Só lembra de rodar ele no modo release)
- Ai no caso é deixar seu app rodando e expor a porta, vai usar um proxy reverso para isto
- Talvez uma solução com Docker e NGinx seja legal
- Resumo: Dá mais trabalho e é mais arriscado do que usar um cloud

Pelo que entendi, há opções através do Kestrel ou Weblistener que já viriam com ASP.NET CORE. Como faria a configuração para um weblistener para o .Net8?
- Sim o Kestrel é o "nativo" e o que recomendo.

Dúvida adicional, conseguiria testar em casa a 'conexão com esse host'? Ou dependeria de ter um IP Fixo para testar? (não sei nada de rede..rsrsrs).
- Depende de um IP que se for fixo é melhor, senão você vai precisar sempre ficar alterando o DNS

**Reply by depaula36** [reply]
Professor, muito obrigado.

**Reply by depaula36** [reply]
Por coincidência. Enquanto esperava a resposta (que veio muito rápida |O| ), lembrei na aula que o sr havia dito que o 'profiles' no launchSettings.json era stand alone e testei colocar no 'applicationUrl' o endereço que aparecia no ipconfig do cmd e depois a porta e deu certo...rsrsrs.....Show demais.

**Reply by depaula36** [reply]
Acessei pelo celular na rede local (era só um teste mesmo). Queria ver se não precisava de servidor externo.

**Reply by andrebaltieri** [reply]
haahhahah show!!!! Que bom que funcionou... desculpa a resposta longa, achei que você queria usar sua máquina como servidor!
